fastcrypto/src/groups/ristretto255.rs

```diff
@@ -98,42 +86,37 @@ impl TryFrom<&[u8]> for RistrettoPoint {
     }
 }
 
-/// Represents a scalar.
-#[derive(
-    Clone,
-    Copy,
-    Debug,
-    Serialize,
-    Deserialize,
-    PartialEq,
-    Eq,
-    From,
-    Add,
-    Sub,
-    Neg,
-    Div,
-    GroupOpsExtend,
-)]
-pub struct RistrettoScalar(ExternalRistrettoScalar);
-
-impl RistrettoScalar {
-    /// Attempt to create a new scalar from the given bytes in canonical representation.
-    pub fn from_canonical_bytes(bytes: [u8; 32]) -> Result<RistrettoScalar, FastCryptoError> {
-        ExternalRistrettoScalar::from_canonical_bytes(bytes)
-            .map_or(Err(FastCryptoError::InvalidInput), |r| {
-                Ok(RistrettoScalar(r))
-            })
+impl ToFromByteArray<RISTRETTO_POINT_BYTE_LENGTH> for RistrettoPoint {
+    fn from_byte_array(bytes: &[u8; RISTRETTO_POINT_BYTE_LENGTH]) -> Result<Self, FastCryptoError> {
+        Self::try_from(bytes.as_slice())
     }
 
-    /// Create a scalar from the low 255 bits of the given 256-bit integer.
-    pub fn from_bits(value: [u8; 32]) -> RistrettoScalar {
-        RistrettoScalar(ExternalRistrettoScalar::from_bits(value))
+    fn to_byte_array(&self) -> [u8; RISTRETTO_POINT_BYTE_LENGTH] {
+        self.compress()
     }
+}
+
+serialize_deserialize_with_to_from_byte_array!(RistrettoPoint);
+generate_bytes_representation!(
+    RistrettoPoint,
+    RISTRETTO_POINT_BYTE_LENGTH,
+    RistrettoPointAsBytes
+);
 
+/// Represents a scalar.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, From, Add, Sub, Neg, Div, GroupOpsExtend)]
+pub struct RistrettoScalar(ExternalRistrettoScalar);
+
+impl RistrettoScalar {
     /// The order of the base point.
     pub fn group_order() -> RistrettoScalar {
         RistrettoScalar(BASEPOINT_ORDER)
     }
+
+    /// Construct a [RistrettoScalar] by reducing a 64-byte little-endian integer modulo the group order.
+    pub fn from_bytes_mod_order_wide(bytes: [u8; 64]) -> Self {
```

Comment:
ooc, why is this not a reference? `bytes: [u8; 64]`

Reply:
You're right. That should definitely be a reference.